File: src/main/java/bad/robot/radiate/teamcity/JsonResponse.java
package bad.robot.radiate.teamcity;

import bad.robot.http.Header;
import bad.robot.http.HttpResponse;
import com.googlecode.totallylazy.Predicate;

import static com.googlecode.totallylazy.Sequences.sequence;

@Deprecated
class JsonResponse {

    private final HttpResponse response;

    public JsonResponse(HttpResponse response) {
        if (!isJson(response))
            throw new UnexpectedContentType(response);
        this.response = response;
    }

    private static boolean isJson(HttpResponse response) {
        if (!sequence(response.getHeaders()).find(json()).isEmpty())
            return true;
        return false;
    }

    private static Predicate<Header> json() {
        return header -> header.name().equalsIgnoreCase("content-type") && header.value().contains("application/json");
    }

    public String body() {
        return response.getContent().asString();
    }
}

File: src/main/scala/bad/robot/radiate/teamcity/JsonResponse.scala
package bad.robot.radiate.teamcity

import simplehttp.{Header, HttpResponse}
import bad.robot.radiate.teamcity.JsonResponse._

object JsonResponse {
  private def isJson(response: HttpResponse): Boolean = {
    import scala.collection.JavaConverters._
    response.getHeaders.asScala.find(json) match {
      case Some(_) => true
      case None => false
    }
  }

  private def json: Header => Boolean = {
    header => header.name().equalsIgnoreCase("content-type") && header.value().contains("application/json")
  }
}

class JsonResponse(response: HttpResponse) {

  if (!isJson(response)) throw new UnexpectedContentType(response)

  def body = response.getContent.asString
}